test/e2e/logforwarding/fluent/crio_split_log_test.go

```diff
@@ -0,0 +1,71 @@
+package fluent_test
+
+import (
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/cluster-logging-operator/pkg/utils"
+	"github.com/openshift/cluster-logging-operator/test/client"
+	"github.com/openshift/cluster-logging-operator/test/helpers/fluentd"
+	"github.com/openshift/cluster-logging-operator/test/helpers/types"
+	"github.com/openshift/cluster-logging-operator/test/matchers"
+	"github.com/openshift/cluster-logging-operator/test/runtime"
+	"path/filepath"
+	rt "runtime"
+	"strings"
+)
+//E2E test for checking correct reassembly after splitting big logs by CRI-O.
+//LogGenerator container will put into stdout prepared huge Java stacktrace in JSON format.
+//CRI-O will spilt it in to the parts, like:
+// 	2021-04-06T00:17:09.669794202Z stdout P First part of JSON log entry
+//	2021-04-06T00:17:09.669794202Z stdout P Second part of JSON log entry
+//	2021-04-06T00:17:10.113242941Z stderr F Last part of JSON log entry
+//After that in should be assembled by fluent-plugin-concat and stored in original state.
+
+var _ = Describe("[ClusterLogForwarder]", func() {
+	var (
+		c *client.Test
+		f *Fixture
+	)
+
+	BeforeEach(func() { c = client.NewTest() })
+	AfterEach(func() { c.Close() })
+
+	Context("with application receiver", func() {
+		It("Checking reassembly after CRI-O log splitting, for big Java stacktrace ", func() {
+			log := getTrace()
+			containerName := "log-generator-" + strings.ToLower(string(utils.GetRandomWord(7)))
+			f = &Fixture{
+				ClusterLogging:      runtime.NewClusterLogging(),
+				ClusterLogForwarder: runtime.NewClusterLogForwarder(),
+				Receiver:            fluentd.NewReceiver(c.NS.Name, "receiver"),
+				LogGenerator: runtime.NewOneLineLogGenerator(c.NS.Name, containerName,
+					strings.ReplaceAll(log, "\"", "\\\"")), // todo
+			}
+			clf := f.ClusterLogForwarder
+			f.Receiver.AddSource(&fluentd.Source{Name: "application", Type: "forward", Port: 24224})
+			addPipeline(clf, f.Receiver.Sources["application"])
+			f.Create(c.Client)
+			r := f.Receiver.Sources["application"].TailReader()
+			line, err := r.ReadLine()
+			matchers.ExpectOK(err)
+			for !strings.Contains(line, fmt.Sprintf("\"container_name\":\"%s\"", containerName)) {
+				line, err = r.ReadLine()
+				println(line)
+				matchers.ExpectOK(err)
+			}
+			// Hack for easy parsing output a proper json array
+			out := "[" + strings.TrimRight(strings.Replace(line, "\n", ",", -1), ",") + "]"
+			logs, err := types.ParseLogs(out)
+			Expect(err).To(BeNil(), "Expected no errors parsing the logs")
+			Expect(logs[0].Message).Should(Equal(log))
+		})
+	})
+})
+
+func getTrace() string {
+	_, filename, _, _ := rt.Caller(0)
+	dir := filepath.Dir(filename)
+	join := filepath.Join(dir, "..", "..", "..", "files/stacktrace.json")
```

Comment:
Consider moving this to a static go file so it is not necessary to load it from disk

Reply:
What is `static go file`, not clear. 
I added this to the file for easy replace content to the new trace

Reply:
@vparfonov Something like https://github.com/openshift/cluster-logging-operator/blob/master/test/helpers/kafka/config.go, just make your JSON stack trace a Go string constant.

Reply:
ok

Reply:
jsonStackTrace can live in its own .go file noone looks at ;)